### PR TITLE
Translate Device Plugins in Chinese

### PR DESCRIPTION
**Pull request by fancc**
Translate Device Plugins in Chinese

### CODE REVIEW

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
+---
+reviewers:
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:2]: pls remove line 2.

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+title: 设备插件
+description: 使用 Kubernetes 设备插件框架来实现适用于 GPU，NIC，FPGA，InfiniBand 以及需要特定于供应商设置的类似资源的插件。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:4]:
```suggestion
description: 使用 Kubernetes 设备插件框架来实现适用于 GPU、NIC、FPGA、InfiniBand 以及类似的需要特定于供应商设置的资源的插件。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+Instead of customizing the code for Kubernetes itself, vendors can implement a
+device plugin that you deploy either manually or as a {{< glossary_tooltip term_id="daemonset" >}}.
+The targeted devices include GPUs, high-performance NICs, FPGAs, InfiniBand adapters,
+and other similar computing resources that may require vendor specific initialization
+and setup.
+-->
+Kubernetes 提供了一个[设备插件框架](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-management/device-plugin.md)，您可以用来将系统硬件资源发布到 {{< glossary_tooltip term_id="kubelet" >}}。
+
+供应商可以实现您手动部署或作为 {{< glossary_tooltip term_id="daemonset" >}} 部署的设备插件，而不必为 Kubernetes 本身定制代码。目标设备包括 GPU、高性能NIC、FPGA、InfiniBand 适配器以及其他类似的计算资源，可能需要特定于供应商的初始化和设置。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:25]:
```suggestion
供应商可以实现设备插件，由您手动部署或作为 {{< glossary_tooltip term_id="daemonset" >}} 来部署，而不必定制 Kubernetes 本身的代码。目标设备包括 GPU、高性能 NIC、FPGA、InfiniBand 适配器以及其他类似的、可能需要特定于供应商的初始化和设置的计算资源。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+
+<!--
+Then, users can request devices in a
+[Container](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core)
+specification as they request other types of resources, with the following limitations:
+
+* Extended resources are only supported as integer resources and cannot be overcommitted.
+* Devices cannot be shared among Containers.
+-->
+然后用户需要去请求其他类型的资源的时候，就可以请求[容器](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core)规范中的设备，但是有以下的限制：
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:86]:
```suggestion
然后用户需要去请求其他类型的资源的时候，就可以在[Container](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core)规范请求这类设备，但是有以下的限制：
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+  * 扩展资源支持整数资源，并且不能被过量使用
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:87]:
```suggestion
  * 扩展资源仅可作为整数资源使用，并且不能被过量使用
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+#
+# This Pod needs 2 of the hardware-vendor.example/foo devices
+# and can only schedule onto a Node that's able to satisfy
+# that need.
+#
+# If the Node has more than 2 of those devices available, the
+# remainder would be available for other Pods to use.
+```
+-->
+假设 Kubernetes 集群正在运行一个设备插件，该插件在一些节点上公布的资源为 `hardware-vendor.example/foo`，下面就是一个 pod 运行请求此资源的示例：
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:116]:
```suggestion
假设 Kubernetes 集群正在运行一个设备插件，该插件在一些节点上公布的资源为 `hardware-vendor.example/foo`。
下面就是一个 Pod 示例，请求此资源以运行某演示负载：
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+        rpc ListAndWatch(Empty) returns (stream ListAndWatchResponse) {}
+
+        // Allocate is called during container creation so that the Device
+        // Plugin can run device specific operations and instruct Kubelet
+        // of the steps to make the Device available in the container
+        rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+  }
+  ```
+
+  * 插件通过 Unix socket 在主机路径 `/ var / lib / kubelet / device-plugins / kubelet.sock` 处向 kubelet 注册自身。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:195]:
```suggestion
  * 插件通过 Unix socket 在主机路径 `/var/lib/kubelet/device-plugins/kubelet.sock` 处向 kubelet 注册自身。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+  * 成功注册自身后，设备插件将以服务模式运行，在此期间，它将持续监控设备运行状况，并在设备状态发生任何变化时向 kubelet 报告。它还负责`分配` gRPC 请求。在`分配`期间，设备插件可能还会做一些设备特定的准备；例如 GPU 清理或 QRNG 初始化。如果操作成功，则设备插件将返回 `AllocateResponse`，其中包含用于访问被分配的设备容器运行时的配置。kubelet将此信息传递到容器运行时。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:196]:
```suggestion
  * 成功注册自身后，设备插件将以服务模式运行，在此期间，它将持续监控设备运行状况，并在设备状态发生任何变化时向 kubelet 报告。它还负责响应 `Allocate` gRPC 请求。在`Allocate`期间，设备插件可能还会做一些设备特定的准备；例如 GPU 清理或 QRNG 初始化。如果操作成功，则设备插件将返回 `AllocateResponse`，其中包含用于访问被分配的设备容器运行时的配置。kubelet 将此信息传递到容器运行时。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+### Handling kubelet restarts
+
+A device plugin is expected to detect kubelet restarts and re-register itself with the new
+kubelet instance. In the current implementation, a new kubelet instance deletes all the existing Unix sockets
+under `/var/lib/kubelet/device-plugins` when it starts. A device plugin can monitor the deletion
+of its Unix socket and re-register itself upon such an event.
+-->
+### 处理 kubelet 重启
+
+一个设备是要能够监测到 kubelet 重启，并且向新的 kubelet 实例来重新注册自己。在当前实现中，当 kubelet 重启的时候，新的 kubelet 实例会删除 `/var/lib/kubelet/device-plugins` 下所有已经存在的 Unix sockets，一个设备插件需要能够监控到它的 Unix socket 被删除，并且当发生此类事件时重新注册自己。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:208]:
```suggestion
设备插件应能监测到 kubelet 重启，并且向新的 kubelet 实例来重新注册自己。在当前实现中，当 kubelet 重启的时候，新的 kubelet 实例会删除 `/var/lib/kubelet/device-plugins` 下所有已经存在的 Unix sockets。设备插件需要能够监控到它的 Unix socket 被删除，并且当发生此类事件时重新注册自己。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+must be mounted as a {{< glossary_tooltip term_id="volume" >}}
+in the plugin's
+[PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core).
+
+If you choose the DaemonSet approach you can rely on Kubernetes to: place the device plugin's
+Pod onto Nodes, to restart the daemon Pod after failure, and to help automate upgrades.
+-->
+## 设备插件部署
+
+你可以降你的设备插件作为你的节点系统的软件包部署为 DaemonSet，或者手动去部署。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:228]:
```suggestion
你可以将你的设备插件作为节点操作系统的软件包来部署、作为 DaemonSet 来部署或者手动部署。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+* Support multiple versions of the device plugin API for backward/forward compatibility.
+
+If you enable the DevicePlugins feature and run device plugins on nodes that need to be upgraded to
+a Kubernetes release with a newer device plugin API version, upgrade your device plugins
+to support both versions before upgrading these nodes. Taking that approach will
+ensure the continuous functioning of the device allocations during the upgrade.
+-->
+## API 兼容性
+
+Kubernetes 设备插件支持还处于 beta 版本。所以在稳定版本出来之前 API 会以不兼容的方式进行更改，作为一个项目，Kubernetes 建议设备插件开发者：
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:250]:
```suggestion
Kubernetes 设备插件支持还处于 beta 版本。所以在稳定版本出来之前 API 会以不兼容的方式进行更改。作为一个项目，Kubernetes 建议设备插件开发者：
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 12]
```diff
@@ -0,0 +1,400 @@
...
+Monitoring agents for device plugin resources can be deployed as a daemon, or as a DaemonSet.
+The canonical directory `/var/lib/kubelet/pod-resources` requires privileged access, so monitoring
+agents must run in a privileged security context.  If a device monitoring agent is running as a
+DaemonSet, `/var/lib/kubelet/pod-resources` must be mounted as a
+{{< glossary_tooltip term_id="volume" >}} in the plugin's
+[PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core).
+
+Support for the "PodResources service" requires `KubeletPodResources` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be enabled. It is enabled by default starting with Kubernetes 1.15.
+-->
+kubelet 提供了gRPC服务来使得正在使用中的设备被发现，并且还未这些设备提供了元数据：
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:297]:
```suggestion
kubelet 提供了 gRPC 服务来使得正在使用中的设备被发现，并且还未这些设备提供了元数据：
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 13]
```diff
@@ -0,0 +1,400 @@
...
+
+```gRPC
+// PodResourcesLister is a service provided by the kubelet that provides information about the
+// node resources consumed by pods and containers on the node
+service PodResourcesLister {
+    rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+}
+```
+
+gRPC服务通过 `/var/lib/kubelet/pod-resources/kubelet.sock` 的 unix socket 来提供服务。设备插件资源的监控代理程序可以部署为守护进程或者 DaemonSet。规范的路径 `/var/lib/kubelet/pod-resources` 需要特权来进入，所以监控代理程序必须要在获得授权的安全的上下文中运行。如果设备监控代理以 DaemonSet 形式运行，必须要在插件的 [PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core) 中声明将 `/var/lib/kubelet/pod-resources` 目录以 {{< glossary_tooltip term_id="volume" >}} 形式被 mount 到容器中。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:307]:
```suggestion
gRPC 服务通过 `/var/lib/kubelet/pod-resources/kubelet.sock` 的 UNIX 套接字来提供服务。设备插件资源的监控代理程序可以部署为守护进程或者 DaemonSet。规范的路径 `/var/lib/kubelet/pod-resources` 需要特权来进入，所以监控代理程序必须要在获得授权的安全的上下文中运行。如果设备监控代理以 DaemonSet 形式运行，必须要在插件的 [PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core) 中声明将 `/var/lib/kubelet/pod-resources` 目录以 {{< glossary_tooltip term_id="volume" >}} 形式被 mount 到容器中。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 14]
```diff
@@ -0,0 +1,400 @@
...
+
+对“PodResources 服务”的支持要求启用“KubeletPodResources”[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)。从 Kubernetes 1.15 开始默认启用。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:309]:
```suggestion
对“PodResources 服务”的支持要求启用 “KubeletPodResources” [特性门控](/docs/reference/command-line-tools-reference/feature-gates/)。从 Kubernetes 1.15 开始默认启用。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+{{% /capture %}}
+{{% capture whatsnext %}}
+
+<!--
+* Learn about [scheduling GPU resources](/docs/tasks/manage-gpus/scheduling-gpus/) using device plugins
+* Learn about [advertising extended resources](/docs/tasks/administer-cluster/extended-resource-node/) on a node
+* Read about using [hardware acceleration for TLS ingress](https://kubernetes.io/blog/2019/04/24/hardware-accelerated-ssl/tls-termination-in-ingress-controllers-using-kubernetes-device-plugins-and-runtimeclass/) with Kubernetes
+* Learn about the [Topology Manager] (/docs/tasks/adminster-cluster/topology-manager/)
+-->
+* 查看 [scheduling GPU resources](/docs/tasks/manage-gpus/scheduling-gpus/) 来学习使用设备插件
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:395]:
```suggestion
* 查看 [调度 GPU 资源](/docs/tasks/manage-gpus/scheduling-gpus/) 来学习使用设备插件
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 16]
```diff
@@ -0,0 +1,400 @@
...
+* 查看在 node 上如何 [advertising extended resources](/docs/tasks/administer-cluster/extended-resource-node/)
+* 阅读如何在 Kubernetes 中如何使用 [hardware acceleration for TLS ingress](https://kubernetes.io/blog/2019/04/24/hardware-accelerated-ssl/tls-termination-in-ingress-controllers-using-kubernetes-device-plugins-and-runtimeclass/)
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:397]: 链接文字未翻译

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 17]
```diff
@@ -0,0 +1,400 @@
...
+{{% /capture %}}
+
+{{% capture body %}}
+
+## 注册设备插件
+
+<!--
+The kubelet exports a `Registration` gRPC service:
+
+```gRPC
```
Comment (tanjunchen) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:36]: 代码段可以不用注释

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 18]
```diff
@@ -0,0 +1,400 @@
...
+-->
+然后用户需要去请求其他类型的资源的时候，就可以请求[容器](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core)规范中的设备，但是有以下的限制：
+  * 扩展资源支持整数资源，并且不能被过量使用
+  * 设备不能在容器之间共享
+
+<!--
+Suppose a Kubernetes cluster is running a device plugin that advertises resource `hardware-vendor.example/foo`
+on certain nodes. Here is an example of a pod requesting this resource to run a demo workload:
+
+```yaml
```
Comment (tanjunchen) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:94]: 这个也可以不用注释 以下类似

**Comment by tanjunchen** [comment]
@fancc 
Thanks for your zh-translation
i think that merge commit is good for  me . see 
image

**Comment by tengqm** [comment]
pls squash your commits.

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 19]
```diff
@@ -0,0 +1,374 @@
...
+<!--
+Then, users can request devices in a
+[Container](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core)
+specification as they request other types of resources, with the following limitations:
+
+* Extended resources are only supported as integer resources and cannot be overcommitted.
+* Devices cannot be shared among Containers.
+-->
+然后用户需要去请求其他类型的资源的时候，就可以在 [Container](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#container-v1-core) 规范请求这类设备，但是有以下的限制：
+  * 扩展资源仅可作为整数资源使用，并且不能被过量使用
```
Comment (gavinfish) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:80]: You need a new line before a list to make it work properly.

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 20]
```diff
@@ -0,0 +1,374 @@
...
+// PodResourcesLister is a service provided by the kubelet that provides information about the
+// node resources consumed by pods and containers on the node
+service PodResourcesLister {
+    rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+}
+```
+
+gRPC 服务通过 `/var/lib/kubelet/pod-resources/kubelet.sock` 的 UNIX 套接字来提供服务。设备插件资源的监控代理程序可以部署为守护进程或者 DaemonSet。规范的路径 `/var/lib/kubelet/pod-resources` 需要特权来进入，所以监控代理程序必须要在获得授权的安全的上下文中运行。如果设备监控代理以 DaemonSet 形式运行，必须要在插件的 [PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core) 中声明将 `/var/lib/kubelet/pod-resources` 目录以 {{< glossary_tooltip term_id="volume" >}} 形式被 mount 到容器中。
+
+对“PodResources 服务”的支持要求启用 “KubeletPodResources” [特性门控](/docs/reference/command-line-tools-reference/feature-gates/)。从 Kubernetes 1.15 开始默认启用。
```
Comment (gavinfish) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:283]:
```suggestion
对“PodResources 服务”的支持要求启用 `KubeletPodResources` [特性门控](/docs/reference/command-line-tools-reference/feature-gates/)。从 Kubernetes 1.15 开始默认启用。
```

Comment (markthink) [reply content/zh/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:283]: @gavinfish  thanks

**Comment by tengqm** [comment]
@fancc Pls squash your commits. [Tips](https://stackoverflow.com/a/43246375).

**Comment by zacharysarah** [comment]
@markthink You merged this PR when it mistakenly contained hundreds of commits, despite clear instructions from other reviewers for the author to squash commits. The consequences of your action have caused significant damage.

**Pull request merged**